configure.ac

```diff
@@ -52,14 +52,24 @@ AC_ARG_WITH([nccl],
 AC_ARG_WITH([mpi],
             AC_HELP_STRING([--with-mpi=PATH], [Path to non-standard MPI installation]),
             [AS_IF([test -d $withval/lib64], [mpi_libdir="lib64"], [mpi_libdir="lib"])
-             CFLAGS="-I$withval/include $CFLAGS"
+             MPIINC="$withval/include"
+             CFLAGS="-I$MPIINC $CFLAGS"
              LDFLAGS="-L$withval/$mpi_libdir $LDFLAGS"],
             [])
 
+
 AC_ARG_ENABLE([tests], [AS_HELP_STRING([--disable-trace],
       [Disable build of test binaries])], [enable_tests=$enableval], [])
 AM_CONDITIONAL([ENABLE_TESTS], [test "x$enable_tests" != "xno"])
 
+# Search first for mpi.h in provided directory, then on system.
+AS_IF([test "x$enable_tests" != "xno" ],
+     [
+       AC_CHECK_HEADERS(["$MPIINC/mpi.h"], [],
```

Comment:
$MPIINC could technically be a relative path, or other things that may confuse AC_CHECK_HEADERS.  The usual path forward is to set CPPFLAGS (as I noted above) and just check for mpi.h as the header.  You also shouldn't need the quotes around mpi.h.